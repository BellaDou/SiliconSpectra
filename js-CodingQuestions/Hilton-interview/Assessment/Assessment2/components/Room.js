import StyledRoom from "./styles/StyledRoom";
import StyledRoomName from "./styles/StyledRoomName";
import StyledRoomTitle from "./styles/StyledRoomTitle";
import StyledSelects from "./styles/StyledSelects";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import {
  checkRoom,
  roomAutoChecks,
  setAdultsOrChildren
} from "../redux/actions/roomActions";

const Room = ({ room, checkRoom, roomAutoChecks, setAdultsOrChildren }) => {
  let option;
  if (!room.ableToCheck || room.checked) {
    option = false;
  } else {
    option = true;
  }

  return (
    <>
      <StyledRoom room={room}>
        <StyledRoomTitle room={room} className="title">
          {room.ableToCheck && (
            <input
              type="checkbox"
              checked={room.checked}
              onChange={() => {
                checkRoom(room.id);
                roomAutoChecks(room.id);
              }}
            />
          )}
          <span>
            <StyledRoomName>Room{room.id}</StyledRoomName>
          </span>
        </StyledRoomTitle>
        <StyledSelects>
          <div>
            <label htmlFor="adults">Adults</label>
            <p>(18+)</p>
            <select
              id="adults"
              disabled={option}
              value={room.adults}
              onChange={e => {
                setAdultsOrChildren(e, room.id);
              }}
            >
              <option value="1">1</option>
              <option value="2">2</option>
            </select>
          </div>
          <div>
            <label htmlFor="children">Children</label>
            <p>(0-17)</p>
            <select
              id="children"
              disabled={option}
              value={room.children}
              onChange={e => {
                setAdultsOrChildren(e, room.id);
              }}
            >
              <option value="0">0</option>
              <option value="1">1</option>
              <option value="2">2</option>
            </select>
          </div>
        </StyledSelects>
      </StyledRoom>
    </>
  );
};
Room.propTypes = {
  room: PropTypes.object.isRequired,
  checkRoom: PropTypes.func.isRequired,
  roomAutoChecks: PropTypes.func.isRequired,
  setAdultsOrChildren: PropTypes.func.isRequired
};
export default connect(null, {
  checkRoom,
  roomAutoChecks,
  setAdultsOrChildren
})(Room);
