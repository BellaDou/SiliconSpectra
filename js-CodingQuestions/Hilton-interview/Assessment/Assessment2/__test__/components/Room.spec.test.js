import React from "react";
import renderer from "react-test-renderer";
import Room from "../../components/Room";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
const mockStore = configureStore([]);

describe("My Connected React-Redux Component", () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      rooms: [
        { id: 1, ableToCheck: false, adults: 1, children: 0 },
        { id: 2, ableToCheck: true, checked: false, adults: 1, children: 0 },
        { id: 3, ableToCheck: true, checked: false, adults: 1, children: 0 },
        { id: 4, ableToCheck: true, checked: false, adults: 1, children: 0 }
      ],
      checkedRooms: []
    });
    component = renderer.create(
      <Provider store={store}>
        {store.getState().rooms.map(room => {
          return <Room key={room.id} room={room} />;
        })}
      </Provider>
    );
  });
  it("should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
  //it("should dispatch an action on button click", () => {});
});
