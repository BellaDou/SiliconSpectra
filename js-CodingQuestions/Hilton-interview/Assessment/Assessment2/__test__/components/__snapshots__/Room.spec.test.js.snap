// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
Array [
  <div
    className="k450ry-0 gHZVtW"
  >
    <div
      className="sc-1ezjdxm-0 hiYShn title"
    >
      <span>
        <h4
          className="giem5x-0 eerikB"
        >
          Room
          1
        </h4>
      </span>
    </div>
    <div
      className="sc-1b9zrps-0 ixQqWJ"
    >
      <div>
        <label
          htmlFor="adults"
        >
          Adults
        </label>
        <p>
          (18+)
        </p>
        <select
          disabled={false}
          id="adults"
          onChange={[Function]}
          value={1}
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
        </select>
      </div>
      <div>
        <label
          htmlFor="children"
        >
          Children
        </label>
        <p>
          (0-17)
        </p>
        <select
          disabled={false}
          id="children"
          onChange={[Function]}
          value={0}
        >
          <option
            value="0"
          >
            0
          </option>
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
        </select>
      </div>
    </div>
  </div>,
  <div
    className="k450ry-0 fIOIaN"
  >
    <div
      className="sc-1ezjdxm-0 cocWes title"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        <h4
          className="giem5x-0 eerikB"
        >
          Room
          2
        </h4>
      </span>
    </div>
    <div
      className="sc-1b9zrps-0 ixQqWJ"
    >
      <div>
        <label
          htmlFor="adults"
        >
          Adults
        </label>
        <p>
          (18+)
        </p>
        <select
          disabled={true}
          id="adults"
          onChange={[Function]}
          value={1}
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
        </select>
      </div>
      <div>
        <label
          htmlFor="children"
        >
          Children
        </label>
        <p>
          (0-17)
        </p>
        <select
          disabled={true}
          id="children"
          onChange={[Function]}
          value={0}
        >
          <option
            value="0"
          >
            0
          </option>
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
        </select>
      </div>
    </div>
  </div>,
  <div
    className="k450ry-0 fIOIaN"
  >
    <div
      className="sc-1ezjdxm-0 cocWes title"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        <h4
          className="giem5x-0 eerikB"
        >
          Room
          3
        </h4>
      </span>
    </div>
    <div
      className="sc-1b9zrps-0 ixQqWJ"
    >
      <div>
        <label
          htmlFor="adults"
        >
          Adults
        </label>
        <p>
          (18+)
        </p>
        <select
          disabled={true}
          id="adults"
          onChange={[Function]}
          value={1}
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
        </select>
      </div>
      <div>
        <label
          htmlFor="children"
        >
          Children
        </label>
        <p>
          (0-17)
        </p>
        <select
          disabled={true}
          id="children"
          onChange={[Function]}
          value={0}
        >
          <option
            value="0"
          >
            0
          </option>
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
        </select>
      </div>
    </div>
  </div>,
  <div
    className="k450ry-0 fIOIaN"
  >
    <div
      className="sc-1ezjdxm-0 cocWes title"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        <h4
          className="giem5x-0 eerikB"
        >
          Room
          4
        </h4>
      </span>
    </div>
    <div
      className="sc-1b9zrps-0 ixQqWJ"
    >
      <div>
        <label
          htmlFor="adults"
        >
          Adults
        </label>
        <p>
          (18+)
        </p>
        <select
          disabled={true}
          id="adults"
          onChange={[Function]}
          value={1}
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
        </select>
      </div>
      <div>
        <label
          htmlFor="children"
        >
          Children
        </label>
        <p>
          (0-17)
        </p>
        <select
          disabled={true}
          id="children"
          onChange={[Function]}
          value={0}
        >
          <option
            value="0"
          >
            0
          </option>
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
        </select>
      </div>
    </div>
  </div>,
]
`;
